src/librustdoc/doctest.rs

```diff
@@ -543,85 +549,100 @@ fn run_test(
         .unwrap_or_else(|| rustc_interface::util::rustc_path().expect("found rustc"));
     let mut compiler = wrapped_rustc_command(&rustdoc_options.test_builder_wrappers, rustc_binary);
 
-    compiler.arg(format!("@{}", doctest.global_opts.args_file.display()));
+    let mut compiler_args = vec![];
```

Comment:
Is there a reason to store these arguments in a temporary variable instead of putting them directly in `compiler`?

Reply:
Yes, because I'm invoking two compilers.

On https://github.com/rust-lang/rust/pull/137899/files#diff-f30ab1f8c0cc2b3a6b7e5f5d2f731800f01774155f534e2a8851187cad09e9fbR613 I build the doctests themselves.

On https://github.com/rust-lang/rust/pull/137899/files#diff-f30ab1f8c0cc2b3a6b7e5f5d2f731800f01774155f534e2a8851187cad09e9fbR665 I build the doctest runner, which is built with unstable features.

Reply:
Then please add a comment explaining this.

Reply:
`conmpiler_args` is the common arguments for both compilers. Regular `compiler` builds the tests, while `compiler_runner` builds the test harness.

The difference is the source path, output path, crate-type, and RUSTC_BOOTSTRAP environment variable.